tools/system_libs.py

```diff
@@ -1896,6 +1907,46 @@ def add_ports_cflags(args, settings):
   return args
 
 
+# Once we require pythonn3.8 we can use shutil.copytree with dirs_exist_ok=True and
+# remove this function.
+def mycopytree(src, dest):
+  with utils.chdir(src):
+    for dirname, dirs, files in os.walk('.'):
+      destdir = os.path.join(dest, dirname)
+      utils.safe_ensure_dirs(destdir)
+      for f in files:
+        shared.safe_copy(os.path.join(src, dirname, f), os.path.join(destdir, f))
+
+
+def install_system_headers():
+  install_dirs = {
+    ('include',): '',
+    ('lib', 'compiler-rt', 'include'): '',
+    ('include', 'compat'): 'compat',
+    ('lib', 'libunwind', 'include'): '',
+    ('lib', 'libc', 'musl', 'arch', 'emscripten'): '',
+    ('lib', 'libcxx'): '',
+    ('include', 'libc'): '',
+    ('include', 'libcxx'): os.path.join('c++', 'v1'),
+    ('lib', 'libcxxabi', 'include'): os.path.join('c++', 'v1'),
+  }
+
+  target_include_dir = shared.Cache.get_include_dir()
+  for src, dest in install_dirs.items():
+    src = shared.path_from_root('system', *src)
+    dest = os.path.join(target_include_dir, dest)
+    mycopytree(src, dest)
+
+  # TODO(sbc): Move these headers back into thier respecive source trees
+  for dirname in ['libc', 'libcxx']:
+    shutil.rmtree(os.path.join(target_include_dir, dirname))
+
+  stamp = shared.Cache.get_path('sysroot_install.stamp')
```

Comment:
I'm unsure what this file does. It doesn't contain the time, so it's not a timestamp?

Reply:
This file simply marks sysroot as having being initially created by copying the copy the system header.    Remove the stamp would cause the headers to be re-installed.  (as would clearing the cache, obviously).

Reply:
sgtm. how about a comment like `The stamp file indicates that the sysroot has been created` ?